### PR TITLE
Updates the action-space of wrappers.RescaleAction

### LINKED ISSUE #568: [Bug Report] `RescaleAction` wrapper does not change the `action_space` accordingly
### Describe the bug

The `RescaleAction` wrapper in (`gymnasium.wrappers.rescale_action`) does not change the `action_space` accordingly.

In contrast, the `RescaleAction` wrapper in `gym.wrappers.rescale_action` does!

When I compare both implementations, the following lines are missing in the `gymnasium` variant compared to the `gym` variant.

```python
self.action_space = spaces.Box(
    low=min_action,
    high=max_action,
    shape=env.action_space.shape,
    dtype=env.action_space.dtype,
)
````

This looks like a bug to me.

### Code example

```shell
import gymnasium as gym
#import gym

env = gym.make("Pendulum-v1")
env = gym.wrappers.rescale_action.RescaleAction(env, min_action=-1, max_action=1)

print(env.action_space)
# gymnasium-->Box(-2.0, 2.0, (1,), float32)
# gym --> Box([-1.], [1.], (1,), float32)
```

### System info

gymnasium: 0.28.1 (pip)
gym: 0.21.0 (pip)
python: 3.7
Ubuntu 20.04

### Additional context

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

### PR DESCRIPTION
**Pull request by mmcaulif**
# Description

Describe the bug
Copy and pasted from issue #568 
> The RescaleAction wrapper in (gymnasium.wrappers.rescale_action) does not change the action_space accordingly.
> 
> In contrast, the RescaleAction wrapper in gym.wrappers.rescale_action does!
> 
> When I compare both implementations, the following lines are missing in the gymnasium variant compared to the gym variant.
> 
> self.action_space = spaces.Box(
>     low=min_action,
>     high=max_action,
>     shape=env.action_space.shape,
>     dtype=env.action_space.dtype,
> )
> This looks like a bug to me.
> 
> Code example
> import gymnasium as gym
> #import gym
> 
> env = gym.make("Pendulum-v1")
> env = gym.wrappers.rescale_action.RescaleAction(env, min_action=-1, max_action=1)
> 
> print(env.action_space)
> # gymnasium-->Box(-2.0, 2.0, (1,), float32)
> # gym --> Box([-1.], [1.], (1,), float32)

Fixes #568

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by mmcaulif** [comment]
> 008 In order to wrap an environment, you must first initialize a base environment. Then you can pass this environment along
> 009 with (possibly optional) parameters to the wrapper's constructor.
> 010 
> 011     >>> import gymnasium as gym
> 012     >>> from gymnasium.wrappers import RescaleAction
> 013     >>> base_env = gym.make("Hopper-v4")
> 014     >>> base_env.action_space
> 015     Box(-1.0, 1.0, (3,), float32)
> 016     >>> wrapped_env = RescaleAction(base_env, min_action=0, max_action=1)
> 017     >>> wrapped_env.action_space
> Expected:
>     Box(-1.0, 1.0, (3,), float32)
> Got:
>     Box(0.0, 1.0, (3,), float32)

The above is result of the failed build action, am I right in saying that the expected behavior outlined is incorrect given the change to RescaleAction? If so, is this due to a mistake on my end?

File: gymnasium/wrappers/__init__.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     Box(-1.0, 1.0, (3,), float32)
     >>> wrapped_env = RescaleAction(base_env, min_action=0, max_action=1)
     >>> wrapped_env.action_space
-    Box(-1.0, 1.0, (3,), float32)
+    Box(0.0, 1.0, (3,), float32)
```
Comment (Kallinteris-Andreas) [inline gymnasium/wrappers/__init__.py:18]: why was this changed?

Comment (mmcaulif) [reply gymnasium/wrappers/__init__.py:18]: https://github.com/Farama-Foundation/Gymnasium/actions/runs/5356895864

It was my understanding that to resolve the error that caused this to fail, I would need to change the docstring in init to accurately reflect the changes made in this PR

Comment (mmcaulif) [reply gymnasium/wrappers/__init__.py:18]: The previous version of line 18 demonstrates the previous functionality where the action space of the env is not updated with the min and max values

Comment (Kallinteris-Andreas) [reply gymnasium/wrappers/__init__.py:18]: nevermind, my bad i misread it

Comment (mmcaulif) [reply gymnasium/wrappers/__init__.py:18]: No worries

**Review by pseudo-rnd-thoughts — APPROVED** [review]
